stable/mongodb/templates/statefulset-primary-rs.yaml

```diff
@@ -68,7 +68,8 @@ spec:
       {{- if .Values.extraInitContainers }}
 {{ tpl .Values.extraInitContainers . | indent 6}}
       {{- end }}
-      {{- if and .Values.volumePermissions.enabled .Values.persistence.enabled }}
+      {{- $needsVolumePermissions := and .Values.volumePermissions.enabled (and ( and .Values.persistence.enabled (not .Values.persistence.existingClaim) ) .Values.securityContext.enabled) }}
```

Comment:
As you are checking here the `securityContext` remove the check frmo the line 88:
```
         {{- if .Values.securityContext.enabled }} <-- this
          securityContext:
            runAsNonRoot: true
            runAsUser: {{ .Values.securityContext.runAsUser }}
           {{- end }} <-- and this
```